docs/reference/mapping/runtime.asciidoc

```diff
@@ -0,0 +1,518 @@
+[[runtime]]
+== Runtime fields
+Typically, you index data into {es} to promote faster search. However, indexing
+can be slow and requires more disk space, and you have to reindex your data to
+add fields to existing documents. With _runtime fields_, you can add
+fields to documents already indexed to {es} without reindexing your data.
+
+[discrete]
+[[runtime-benefits]]
+=== Benefits
+Because runtime fields aren't indexed, you can more quickly ingest raw data
+into the Elastic Stack and access it immediately. By dynamically evaluating
+runtime fields at search time, you can optimize disk space by choosing which
+fields to index. If you later decide that a runtime field is increasingly used
+for filtering or aggregation, you can add the field to `_source` to gain faster
+search speed.
+
+Instead of reindexing your data to add fields, add runtime fields to the
+mapping definition. You can access runtime fields from the search API like any
+other field, and {es} sees runtime fields no differently.
+
+Alternatively, you can specify a `runtime_mappings` section in a search request
+to use runtime fields that exist only as part of the query. This
+flexibility allows you to create or shadow field values in `_source` for the
```

Comment:
shadow existing fields? Shadowing is not specific to fields in _source, and actually from a search request you could even shadow an existing runtime field defined in the index mappings.

Reply:
I updated the shadowing section, which discusses how to use shadowing in different capacities.